specification.json

```diff
@@ -358,6 +365,28 @@
             "RFC 2119 keyword": "MUST",
             "children": []
         },
+        {
+            "id": "Requirement 2.1.2",
+            "machine_id": "requirement_2_1_2",
+            "content": "The provider MAY define an `allFlagMetadata` field or accessor of type collection of flag_metadata, which identifies all available flag keys in the provider.",
+            "RFC 2119 keyword": "MAY",
+            "children": []
+        },
+        {
+            "id": "Condition 2.1.3",
+            "machine_id": "condition_2_1_3",
+            "content": "The implementing language has an asynchronous support.",
+            "RFC 2119 keyword": null,
+            "children": [
+                {
+                    "id": "Conditional Requirement 2.1.3.1",
+                    "machine_id": "conditional_requirement_2_1_3_1",
+                    "content": "The accessor for `allFlagMetadata` SHOULD be asynchronous.",
+                    "RFC 2119 keyword": "SHOULD",
+                    "children": []
+                }
+            ]
+        },
```

Comment:
I'm not 100% sure this is necessary to specify. I think most language implementations will make the right choice here, especially if you mentioned something like this in a non-normative section.

Reply:
I was on the fence about making this a full requirement. I like the idea of doing this as non-normative text 👍🏻